luigi/contrib/sqla.py

```diff
@@ -130,9 +130,25 @@ def requires(self):
 import datetime
 import itertools
 import sqlalchemy
-
+from threading import Lock
 logger = logging.getLogger('luigi-interface')
 
+_engine_hash={}
+_mutex = Lock()
+def _get_engine(connection_string, echo):

```

Comment:
would prefer to put this in a classmethod on SQLAlchemyTarget and use class variables instead of global variables